torchrl/envs/transforms/transforms.py

```diff
@@ -2429,3 +2429,50 @@ def __repr__(self) -> str:
             f"{self.__class__.__name__}(decay={self.decay:4.4f},"
             f"eps={self.eps:4.4f}, keys={self.in_keys})"
         )
+
+
+class StepCounter(Transform):
+    """Counts the steps from a reset and sets the done state to True after a certain number of steps.
+
+    Args:
+        max_steps (:obj:`int`, optional): the maximum number of steps to take before setting the done state to True.
+    """
+
+    invertible = False
+    inplace = True
+
+    def __init__(self, max_steps: Optional[int]):
+        self.max_steps = max_steps
+        super().__init__([])
+
+    def reset(self, tensordict: TensorDictBase) -> TensorDictBase:
+        tensordict.set(
```

Comment:
If we want to make this work in parallel settings, we can work with the "reset_workers" key, which indicates what are the workers that need to be reset